tests/SerdeTestCases.php

```diff
@@ -1059,6 +1059,27 @@ public function datetime_fields_support_custom_output_format(): void
         self::assertEquals($expected, $result);
     }
 
+    #[Test]
+    public function datetime_field_can_be_datetimeinterface(): void
```

Comment:
I think this test would work better as a case in `round_trip_examples()`.  Most tests are using concrete classes rather than anonymous, as... I'm kinda surprised it even works this way. :smile:

Reply:
Honestly, if I hadn't spun my wheels for so long doing my own (de)serializing code, I wouldn't have known either 😅 I'll get this change in shortly.

Reply:
Change is in; LMK if I should squash the commits